### PR TITLE
Updates and fixes

### PR DESCRIPTION
**Pull request by ljakimczuk**
Updates and fixes introduced in this PR:

* migration from RSA to ECC keys. Users who have been using RSA can continue doing that, however the `kgs` automation is going to create ECC keys only, so this is basically a follow up.
* make `mapi` mandatory. So far it was optional, however we agreed that for automation it would be much easier to work with a fixed structure, so it needs to be emphasized in the docs and examples as well now. Users who haven't been using it, may cointinue doing that. The `kgs` is going to introduce the `--skip-mapi` flag that would result in omitting the `mapi`. We should however clearly note in the docs that `mapi` is obligatory from now on, and if possible talk the current GitOps users into migrating their structures. Ideally we drop the `--skip-mapi` at some point.
* so far `mapi` and `direct` Kustomization CRs have been kept in a single file, this PR splits them. Both Kustomization live in a different namespaces and hence must also use different secrets for decryption, so it is much easier and natural to work with them as separate files, especially from the automation perspective.
* restore fixes introduced [in this PR](https://github.com/giantswarm/gitops-template/pull/58/files#) that get reverted at some point by a mistake I think. The linked PR standardized the structure by moving cluster with apps, aka `environments` into the `cluster` directory, so that we do not need to consider it as a special case from the automation perspective. Users can just create it as a regular cluster from a base. Technically, as justified in the linked PR, cluster with apps is still a cluster, and hence should be configured under cluster definition directory.
* fixes to naming. Mainly in the `repo_structure.md` I noticed inconsistencies, sometimes `CAPI_WC_NAME` and sometimes `WC_NAME` were used. I tried to move it to `CAPI_WC_NAME`, and I did the same for the examples we have. I spotted some of these inconsistencies across the docs, tried to fix them.
* removes `CAPI_WC_NAME_NO_OUT_OF_BAND` as being redundant. `CAPI_WC_NAME` already provides an example without out-of-band delivery.

ToDos:
- [ ] re-encrypt secrets with master ECC
- [ ] update fake flux build to use actual `flux build`

### CODE REVIEW

File: docs/add_wc_structure.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ This includes:
 
 ## Example
 
-An example of a WC cluster directory structure is available in [WC_NAME/](/management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/).
+An example of a WC cluster directory structure is available in [WC_NAME/](/management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/CAPI_WC_NAME/).
```
Comment (piontec) [inline docs/add_wc_structure.md:24]:
```suggestion
An example of a WC cluster directory structure is available in [CAPI_WC_NAME/](/management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/CAPI_WC_NAME/).
```

File: docs/repo_structure.md [hunk 2]
```diff
@@ -42,12 +43,13 @@ management-clusters
             └── [workload-clusters]
                 ├── kustomization.yaml
                 ├── CAPI_WC_NAME.yaml
+                ├── [CAPI_WC_NAME-direct.yaml]
                 └── CAPI_WC_NAME                             managed from MC_NAME.yaml
 -----------------------------------------------------------------------------------
-                    ├── [out-of-band]                  WC_NAME.yaml responsibility
+                    ├── [out-of-band]                     CAPI_WC_NAME.yaml responsibility
```
Comment (piontec) [inline docs/repo_structure.md:49]: Same here: if we're hinting "any name of a WC", should we just use `MC_NAME.yaml` here?

File: docs/repo_structure.md [hunk 3]
```diff
@@ -398,28 +397,31 @@ management-clusters
...
+hence tells Flux to create, only the `CAPI_WC_NAME.yaml` from the `workload-clusters` directory, see example below:
 
 ```text
 resources:
-- WC_NAME.yaml
+- CAPI_WC_NAME.yaml
 ```
 
-The `WC_NAME.yaml` Kustomization CR, when created, points to the respective `WC_NAME/` directory and reconciles
+The `WC_NAME.yaml` Kustomization CR, when created, points to the respective `CAPI_WC_NAME/` directory and reconciles
```
Comment (piontec) [inline docs/repo_structure.md:418]: This sounds inconsistent now - again `CAPI` prefix issue

**Review by piontec — COMMENTED** [review]
Nice and a ton of improvements :)

I have just two things:

1) Tests are failing - one of the Kustomizations doesn't come up OK (or should be added to the list of expected-to-fail Kustomizations, check README.md)

2) I was trying to make one differentiation in the docs. Wherever I was using `CAPI_WC_NAME` I had in mind the specific cluster configuration and its files that we have in our examples in `management_clusters` dir. Where I was using `WC_NAME`, I was hinting that it's just "a WC name", a placeholder a user should replace. Apparently, I failed to make it consistent or just write about this naming convention - now I see it confused you as well :| WDYT about it? Should we make it explicit and use this convention?

**Comment by ljakimczuk** [comment]
> I was trying to make one differentiation in the docs. Wherever I was using CAPI_WC_NAME I had in mind the specific cluster configuration and its files that we have in our examples in management_clusters dir. Where I was using WC_NAME, I was hinting that it's just "a WC name", a placeholder a user should replace. Apparently, I failed to make it consistent or just write about this naming convention - now I see it confused you as well :| WDYT about it? Should we make it explicit and use this convention?

Ok, so if I get it right, it a bit confusing because `CAPI_WC_NAME` was used in the repository tree doc, where according to what you've said, it should rather be `WC_NAME` there, as we do not reference any examples there, we merely present the structure, with placeholders that should be replaced. I think we should pick only one way of naming, because otherwise it will be very confusing and error prone. Since this structure is about CAPI clusters I think we can stick to `CAPI_`, WDYUT?

**Comment by ljakimczuk** [comment]
> Since this structure is about CAPI clusters I think we can stick to CAPI_, WDYUT?

Ok, I take it back, there are too many places where we whould need to change it, it's better to stick to `WC_NAME` which has a longer history in this repo.

File: docs/add_wc_structure.md (hunk 1 above)
Comment (ljakimczuk) [reply docs/add_wc_structure.md]: Changed it back to `WC_NAME`

File: docs/repo_structure.md [hunk 4]
```diff
@@ -31,7 +31,8 @@ management-clusters
     ├── [OTHER_RESOURCES]
     ├── secrets
     |   ├── MC_NAME.gpgkey.enc.yaml
-    |   └── [WC_NAME.gpgkey.enc.yaml]
+    |   ├── [CAPI_WC_NAME.gpgkey.enc.yaml]
```
[the remark this answers is not in the document]
Comment (ljakimczuk) [reply docs/repo_structure.md]: Ok, I changed it back to the `WC_NAME`, let's try to keep them as simple as possible and hence try to not use variables that seem interchangeable for different purposes.

**Review by piontec — APPROVED** [review]
LGTM! Good idea with removing `origin` annotation from expectations!

**Pull request merged**